### PR TITLE
build: Set minimum required Windows version to Windows 7

### PR DESCRIPTION
**Pull request by str4d**
Cherry-picked from the following upstream PRs:
- bitcoin/bitcoin#14922
- bitcoin/bitcoin#17756
- bitcoin/bitcoin#17740
- bitcoin/bitcoin#18003
- bitcoin/bitcoin#18928
- bitcoin/bitcoin#18956
- bitcoin/bitcoin#19375
- bitcoin/bitcoin#15704

Also adds `libbcrypt` to the list of linked libraries, as Boost starts
depending on `BCryptGenRandom` (as will `getrandom 0.2` once we pull the
latest Rust crates in).

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -4,3 +4,8 @@ release-notes at release time)
 Notable changes
 ===============
 
+Build system changes
+--------------------
+
+- Cross-compilation support for Windows XP and Vista has been removed.
+  Cross-compiled Windows binaries now target a minimum of Windows 7.
```
Comment (daira) [inline doc/release-notes.md:11]:
```suggestion
  Cross-compiled Windows binaries now target a minimum of Windows 7 (64-bit only).
```

**Review by daira — APPROVED** [review]
This builds for me with the following warnings (not sure they are due to this PR; I think I've seen them before):
```
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_uncaught_exceptions (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_decrement_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_increment_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_current_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_rethrow_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_uncaught_exceptions (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_decrement_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_increment_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_current_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_rethrow_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_uncaught_exceptions (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_decrement_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_increment_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_current_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_rethrow_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_uncaught_exceptions (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_decrement_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_increment_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_current_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_rethrow_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: gtest/zcash_gtest-test_checktransaction.o: locally defined symbol imported: __cxa_demangle (defined in libc++abi.a(cxa_demangle.cpp.obj)) [LNK4217]
lld-link: warning: gtest/zcash_gtest-test_zip32.o: locally defined symbol imported: __cxa_demangle (defined in libc++abi.a(cxa_demangle.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_uncaught_exceptions (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_decrement_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_increment_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_current_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_rethrow_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_uncaught_exceptions (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_decrement_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_increment_exception_refcount (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_current_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libc++.a(exception.cpp.obj): locally defined symbol imported: __cxa_rethrow_primary_exception (defined in libc++abi.a(cxa_exception.cpp.obj)) [LNK4217]
lld-link: warning: libboost_unit_test_framework.a(execution_monitor.obj): locally defined symbol imported: __cxa_demangle (defined in libc++abi.a(cxa_demangle.cpp.obj)) [LNK4217]```

LNK4217 is https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4217?view=msvc-160 , which doesn't seem critical.

File: doc/release-notes.md (hunk 1 above)
Comment (str4d) [inline doc/release-notes.md:11]:
```suggestion
- Cross-compilation support for Windows XP, Windows Vista, and 32-bit Windows
  binaries, has been removed. Cross-compiled Windows binaries are now 64-bit
  only and target a minimum of Windows 7.
```

**Comment by str4d** [comment]
Yep, those warnings were there before this PR.

@zkbot r+

**Comment by daira** [comment]
I'm trying something to get rid of the link warnings.

**Comment by str4d** [comment]
[CI has passed for all required builders](https://ci.z.cash/#/builders/1/builds/655), and this PR would have merged several hours ago, but for a bug in CI that is causing it to hang on two non-required builders. I'm clicking the merge button.

**Pull request merged**